### ISSUE TITLE
mclapply + readr to read in large files yields long vectors not supported error

### ISSUE THREAD

**Issue by SummerZheng** [hadley/readr#319]
When I tried to read in multiple large csv files in parallel 
```
df_list <- parallel::mclapply(files, function(file) {
  readr::read_csv(file.path(file_dir, file))
}, mc.cores = length(files)))
```

it throws the following errors after individual file read in complete
```
Error in sendMaster(try(lapply(X = S, FUN = FUN, ...), silent = TRUE)) :
  long vectors not supported yet: fork.c:378
Package ‘bettertrace’ overwriting ‘options("error")’
```
And I have 10 files, each of size ~ 500MB, and contains a data frame of ~150,000 rows * 1800 columns. 
This error happens for readr version `0.2.2`, if I pass in  `n_max = 10000`, or use readr version '0.1.1', the error disappears. So I wonder what changes from 0.1.1 to 0.2.2 could cause this memory issue.

Many thanks!

**Comment by dselivanov** [comment]
**offtop**. Generally reading files in parallel is not good idea, because usually bottleneck is disk, not cpu.

**Comment by hadley** [comment]
Please provide a reproducible example

**Comment by hadley** [comment]
Oh this is probably just a duplicate of #309
